### ISSUE TITLE
Column-wise CSV writing of array_t

### ISSUE THREAD

**Issue by EightSQ** [stephenberry/glaze#253]
ccb1be4a0dc65d79b7c9757dcd130889bc4e2579 removed the `glz::context::csv_index` variable, however, in main there is a reference remaining [here](https://github.com/stephenberry/glaze/blob/b52a7d56f17bf3f53d9c2288f59d4acff9cb4e9f/include/glaze/csv/write.hpp#L86). This leads to a compilation error when CSV-writing a resizeable `array_t` (like `std::vector<>`) with the `colwise` layout (see below).

<details><summary>Compilation error</summary>
<p>

```
/build/_deps/glaze-src/include/glaze/csv/write.hpp:86:50: error: no member named 'csv_index' in 'glz::context'
                  write<csv>::op<Opts>(value[ctx.csv_index], ctx, b, ix);
                                             ~~~ ^
/build/_deps/glaze-src/include/glaze/csv/write.hpp:26:47: note: in instantiation of function template specialization 'glz::detail::to_csv<std::vector<Record>>::op<opts{10000, false, true, true, true, false, 32, 3, false, true, true, false, false, 1, false, false}, std::string &, std::vector<Record> &, glz::context &, unsigned long &>' requested here
            to_csv<std::decay_t<T>>::template op<Opts>(std::forward<T>(value), std::forward<Ctx>(ctx),
                                              ^
/build/_deps/glaze-src/include/glaze/core/write.hpp:40:44: note: in instantiation of function template specialization 'glz::detail::write<10000>::op<opts{10000, false, true, true, true, false, 32, 3, false, true, true, false, false, 1, false, false}, std::vector<Record> &, glz::context &, std::string &, unsigned long &>' requested here
      detail::write<Opts.format>::template op<Opts>(std::forward<T>(value), ctx, buffer, ix);
                                           ^
/build/_deps/glaze-src/include/glaze/core/write.hpp:50:7: note: in instantiation of function template specialization 'glz::write<opts{10000, false, true, true, true, false, 32, 3, false, true, true, false, false, 1, false, false}, std::vector<Record> &, std::string, glz::context &>' requested here
      write<Opts>(std::forward<T>(value), buffer, ctx);
      ^
/build/_deps/glaze-src/include/glaze/csv/write.hpp:325:7: note: in instantiation of function template specialization 'glz::write<opts{10000, false, true, true, true, false, 32, 3, false, true, true, false, false, 1, false, false}, std::vector<Record> &, std::string>' requested here
      write<opts{.format = csv, .layout = layout}>(std::forward<T>(value), buffer);
      ^
/src/generic_generateable.hpp:371:33: note: in instantiation of function template specialization 'glz::write_file_csv<1U, std::vector<Record> &>' requested here
    glz::write_error err = glz::write_file_csv<glz::colwise>(records, file_name);
```

</p>
</details>

**Comment by stephenberry** [comment]
Thanks for pointing this issue out. Will get to this soon.

**Comment by stephenberry** [comment]
@EightSQ
I removed the old code and instead added a static_assert (#256). I don't know if your compiler was just complaining about the deprecated variable, or if your use case tried to compile that constexpr branch.

Currently we don't support dynamic arrays within dynamic arrays, because this poses performance problems for loading row-wise documents.

Do you need this behavior? If so, could you give an example of what you're trying to achieve.

Thanks!

**Comment by EightSQ** [comment]
@stephenberry At the time I came across this issue I was trying to achieve something I think is not possible at the moment (and I did not read the manual properly at first).

What I tried to do was CSV serialization based on an "array" of structs, i.e. one array element = struct instance creates one CSV row. I found a way to modify glaze to make it work for me but I have not found a "nice" solution that I can share. And it doesn't support array's at all.
